### DISCUSSION TITLE
seqr on GKE

### DISCUSSION THREAD

**Discussion by hpuri** [broadinstitute/seqr#2909] [Q&A]
I was able to run using docker-compose on GCP , now trying GKE 
I am stuck at running the servctl script which generates the following error message - any suggestions on how to get past it ?
There is a related issue of whether this is python 2 vs 3 , assuming 3
Thanks

```
python servctl deploy-all gcloud-prod
2022-08-22 21:31:24,235 INFO     ==> deploy/kubectl_helpers/utils/check_context.sh prod
/usr/lib/python3.9/subprocess.py:941: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used
  self.stdout = io.open(c2pread, 'rb', bufsize)
Traceback (most recent call last):
  File "/home/admin_hmp_altostrat_com/seqr/servctl", line 79, in <module>
    deploy(args.deployment_target, components=components_to_deploy, runtime_settings=runtime_settings)
  File "/home/admin_hmp_altostrat_com/seqr/deploy/servctl_utils/deploy_command_utils.py", line 215, in deploy
    run('deploy/kubectl_helpers/utils/check_context.sh {}'.format(deployment_target.replace('gcloud-', '')))
  File "/home/admin_hmp_altostrat_com/seqr/deploy/servctl_utils/shell_utils.py", line 61, in run
    if out.endswith('\r') or (out.endswith('\n') and previous_is_slash_r):
TypeError: endswith first arg must be bytes or a tuple of bytes, not str
```

**Comment by hanars** [comment]
GKE is not supported for local installations at this time. While you are welcome to write your own configurations to deploy seqr on GKE, the kubectl_helpers and other scripts that are left out of all the documentation for local installations are unable to be used for local installations. Our team does not have the bandwidth to help users who are running any deployment other than recommended and supported docker compose installation

**Comment by hpuri** [comment]
I appreciate the response and sympathize on the lack of resources/bandwidth.
Is there a design document / architecture that my team can look at and identify components that need to change ?

**Comment by hanars** [comment]
I'm sorry, at this time we have no documentation or support for deploying to GKE. As an FYI, the scripts in the repo do not contain enough information/ functionality to successfully deploy seqr, even if you get them to run so I really do not recommend that you use them. If you are determined to run GKE, I would recommend you look at the components defined in the docker compose file and write your own kubernetes manifests for each them
